src/main/host/descriptor/pipe.rs

```diff
@@ -43,6 +43,16 @@ impl PipeFile {
         self.flags = flags;
     }
 
+    pub fn close(&mut self, event_queue: &mut EventQueue) -> SyscallReturn {
+        // set the closed flag and remove the active flag
+        self.copy_status(
+            FileStatus::CLOSED | FileStatus::ACTIVE,
```

Comment:
Why do we both remove and add the CLOSED flag?

Reply:
I'm not sure what you mean here. This sets the mask for the `copy_status` operation as `FileStatus::CLOSED | FileStatus::ACTIVE`, and sets the value under that mask to `FileStatus::CLOSED`. So this adds the `CLOSED` flag, and removes the `ACTIVE` flag.`

Reply:
Ah ok. `copy_status` is a little difficult to follow, especially at the call sites, but I'm not sure how to improve it.

Reply:
Yeah I agree, and I'm not sure how to improve it either. I could set the flags separately, but currently it would require two calls to `handle_status_change()`, which would then emit two events.